packages/explorer/src/app/(explorer)/[chainName]/worlds/[worldAddress]/explore/TableColumn/useSetField.tsx

```diff
@@ -0,0 +1,78 @@
+import { ExternalLinkIcon } from "lucide-react";
+import { useParams } from "next/navigation";
+import { toast } from "sonner";
+import { Hex } from "viem";
+import { useConfig } from "wagmi";
+import { waitForTransactionReceipt, writeContract } from "wagmi/actions";
+import { Table } from "@latticexyz/config";
+import {
+  encodeField,
+  getFieldIndex,
+  getSchemaPrimitives,
+  getSchemaTypes,
+  getValueSchema,
+} from "@latticexyz/protocol-parser/internal";
+import IBaseWorldAbi from "@latticexyz/world/out/IBaseWorld.sol/IBaseWorld.abi.json";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { useChain } from "../../../../../hooks/useChain";
+import { blockExplorerTransactionUrl } from "../../../../../utils/blockExplorerTransactionUrl";
+
+type Props<table extends Table, fieldName extends keyof getValueSchema<table>> = {
+  table: table;
+  keyTuple: readonly Hex[];
+  fieldName: fieldName;
+};
+
+export function useSetField<
+  table extends Table,
+  fieldName extends keyof getValueSchema<table> & string,
+  value extends getSchemaPrimitives<getValueSchema<table>>[fieldName] = getSchemaPrimitives<
+    getValueSchema<table>
+  >[fieldName],
+>({ table, keyTuple, fieldName }: Props<table, fieldName>) {
```

Comment:
turns out all the type fun that I had didn't matter because we don't have specific types in this context (just a generic `Table` and `string` field name

but leaving this in because it might be useful to extract out into a place where we do have stronger input types